spotipy/client.py

```diff
@@ -1481,3 +1562,28 @@ def _get_id(self, type, id):
 
     def _get_uri(self, type, id):
         return "spotify:" + type + ":" + self._get_id(type, id)
+
+    def _search_multiple_markets(self, q, limit, offset, type, markets, total):
+        results = {
+                type + 's': {
+                    'href': [],
+                    'items': [],
+                    'limit': limit,
+                    'next': None,
+                    'offset': 0,
+                    'previous': None,
+                    'total': 0
+                }
+            }
+        for country in markets:
+            result = self._get(
+                "search", q=q, limit=limit, offset=offset, type=type, market=country
+            )
+            results[type + 's']['href'].append(result[type + 's']['href'])
+            results[type + 's']['items'] += result[type + 's']['items']
+            results[type + 's']['total'] += result[type + 's']['total']
+            if total and len(results[type + 's']['items']) >= total:
+                # splice 'items' to only include number of results requested
+                results[type + 's']['items'] = results[type + 's']['items'][:total]
+                return results
```

Comment:
It seems weird to me here that we're messing with the API response returned by Spotify. This could be confusing at times.

I think a better choice would be to have a key for each market in the result. This way we do not mess with the internal response returned by Spotify. The results would then look something like below on using `sp.search`:

```python
>>> result = sp.search("lany good guys", market=["US", "GB"])
>>> us_tracks = result["US"]["tracks"]
>>> us_tracks.keys()
dict_keys(['href', 'items', 'limit', 'next', 'offset', 'previous', 'total'])

>>> us_tracks["href"]
'https://api.spotify.com/v1/search?query=lany+good+guys&type=track&market=US&offset=0&limit=10'

>>> gb_tracks = result["GB"]["tracks"]
>>> gb_tracks["href"]
'https://api.spotify.com/v1/search?query=lany+good+guys&type=track&market=GB&offset=0&limit=10'
```

@stephanebruckert what do you think?

Reply:
I completely agree - It feels hacky. I was actually thinking of implementing the market key idea you mention as I was working on it, but I wanted to try and preserve the format of the response from Spotify.

Reply:
Sorry I merged this too quickly. I agree, it does feel weird. Now that you point it out it's true that mixing up results does not make sense, mainly due to the fact that results are supposed to be ordered by "relevancy".

The reason I thought it was OK in the first place is that in many cases, we are only interested in the first result. In this specific case, getting a dictionary of all countries can make it difficult to find in which country the release was found.

But it's probably nicer to let developers handle country results the way they want, since we don't really know what their use case is. So I would be all for this change

Reply:
@stephanebruckert It's all good. Thank you for maintaining spotipy all this while! I agree that devs should be able to do whatever they want with the country results later on.

@NLeRoy917 Are you willing to make a new PR for this?

Reply:
Yes, definitely - I can work on that